changelog.d/13694.bugfix

```diff
@@ -0,0 +1 @@
+Avoid calculating the unstable unread counts from [MSC2654](https://github.com/matrix-org/matrix-spec-proposals/pull/2654) unless the results will be returned to the user.
```

Comment:
This isn't a great changelog, I don't know if this is a bugfix or feature or misc or what. 😢

Reply:
Yeah that's a tricky one. Maybe a `feature` since it's a functional and client-facing change to an existing feature? Or a keep it as a bugfix but make it more clear what the bug is (e.g. "Fix a bug introduced in Synapse x.xx.x that would cause the unstable unread counts from MSC2654 to be calculated even if the feature is disabled").

Also I'm unclear on what "returned to the user" means in this case. Does it mean sent to the client, shown to the user by the client, ...?

Reply:
> Also I'm unclear on what "returned to the user" means in this case. Does it mean sent to the client, shown to the user by the client, ...?

Sent to the client in syncs, yes.